darts/models/torch_forecasting_model.py

```diff
@@ -200,6 +205,131 @@ def __init__(self,
         self.lr_scheduler_cls = lr_scheduler_cls
         self.lr_scheduler_kwargs = dict() if lr_scheduler_kwargs is None else lr_scheduler_kwargs
 
+        checkpoints_folder = _get_checkpoint_folder(self.work_dir, self.model_name)
+        if os.path.exists(checkpoints_folder) and len(glob(os.path.join(checkpoints_folder, "checkpoint_*"))) > 0:
+            self.load()
+
+    def load(self, best=False):
+        ''' Loads the model from a checkpoint by given name and work directory.
+
+            Parameters
+            ----------
+            best
+                Load the best performing checkpoint checkpoint instead of the latest checkpoint (the default)
+
+        '''
+        params = {k:v for k,v in self.get_params() if k not in self.immutable_args}
+        model = self.load_from_checkpoint(self.model_name, self.work_dir, best)
+        self.__dict__.update(model.__dict__)
+        self.set_params(**params)
+
+    @classmethod
+    def _get_param_names(cls):
```

Comment:
This code seems to be used only from within `get_params()`. Could you maybe copy this method over into `get_params()` ?
Also please adapt all the sklearn things (and remove what's unnecessary).